lib/routemap.c

```diff
@@ -49,6 +49,8 @@ DEFINE_QOBJ_TYPE(route_map)
 #define IPv6_PREFIX_LIST "ipv6 address prefix-list"
 #define IPv4_MATCH_RULE "ip "
 #define IPv6_MATCH_RULE "ipv6 "
+#define IPv4_NH_MATCH_RULE "ip next-hop"
```

Comment:
if we are making this a define then we should convert all the code that passes in this to the route-map code. ( see bgp_routemap.c at a min ).  Probably should be it's own commit

Reply:
As part of the optimization during the route-map processing, I had initially done the following:
* An IPv4 route prefix would not be evaluated against a route-map index if it has ANY IPv6 match clause.
* Similarly, an IPv6 route prefix would not be evaluated against a route-map index if it has ANY IPv4 match clause.
I had not taken into consideration IPv4-over-IPv6 where IPv4 route prefixes could potentially match against IPv6 next-hop match clause.
Similarly, I had also not taken into consideration IPv6-over-IPv4.

In the first patchset, I had considered the IPv4 and IPv6 next-hop match clauses as exceptions.
This exception would ensure that all routes would be evaluated against route-map indices having IPv4 and/or IPv6 next-hop match clauses.

Such exceptions, however, could be difficult to maintain in the future if newer route-map match clauses are introduced. 
Considering this, I’ve now programmed the evaluation of IPv4 route prefixes against route-map indices if-and-only-if those route-map indices have the IPv6 prefix-list match clause and vice versa. That way, there will be no need to add any exception rules in the future.